drivers/spi/spi_nrfx_spim.c

```diff
@@ -184,6 +196,91 @@ static int configure(const struct device *dev,
 	return 0;
 }
 
+#ifdef CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58
+/*
+ * Brief Workaround for transmitting 1 byte with SPIM.
+ *
+ * Derived from the setup_workaround_for_ftpan_58() function from
+ * the nRF52832 Rev 1 Errata v1.6 document anomaly 58 workaround.
+ *
+ * Warning Must not be used when transmitting multiple bytes.
+ *
+ * Warning After this workaround is used, the user must reset the PPI
+ * channel and the GPIOTE channel before attempting to transmit multiple
+ * bytes.
+ */
+static void anomaly_58_workaround_setup(const struct device *dev)
+{
+	struct spi_nrfx_data *dev_data = get_dev_data(dev);
+	const struct spi_nrfx_config *dev_config = get_dev_config(dev);
+	NRF_SPIM_Type *spim = dev_config->spim.p_reg;
+	uint32_t ppi_ch = dev_data->ppi_ch;
+	uint32_t gpiote_ch = dev_data->gpiote_ch;
+	uint32_t eep = (uint32_t)&NRF_GPIOTE->EVENTS_IN[gpiote_ch];
+	uint32_t tep = (uint32_t)&spim->TASKS_STOP;
+
+	dev_data->anomaly_58_workaround_active = true;
+
+	/* Create an event when SCK toggles */
+	nrf_gpiote_event_configure(NRF_GPIOTE, gpiote_ch, spim->PSEL.SCK,
+				   GPIOTE_CONFIG_POLARITY_Toggle);
+	nrf_gpiote_event_enable(NRF_GPIOTE, gpiote_ch);
+
+	/* Stop the spim instance when SCK toggles */
+	nrf_ppi_channel_endpoint_setup(NRF_PPI, ppi_ch, eep, tep);
+	nrf_ppi_channel_enable(NRF_PPI, ppi_ch);
+
+	/* The spim instance cannot be stopped mid-byte, so it will finish
+	 * transmitting the first byte and then stop. Effectively ensuring
+	 * that only 1 byte is transmitted.
+	 */
+}
+
+static void anomaly_58_workaround_clear(struct spi_nrfx_data *dev_data)
+{
+	uint32_t ppi_ch = dev_data->ppi_ch;
+	uint32_t gpiote_ch = dev_data->gpiote_ch;
+	nrfx_err_t err_code;
+
+	if (dev_data->anomaly_58_workaround_active) {
+		err_code = nrfx_ppi_channel_disable(ppi_ch);
+		if (err_code != NRFX_SUCCESS) {
+			LOG_ERR("Failed to disable PPI channel: %u", err_code);
+		}
```

Comment:
I think it would be better to use `nrf_ppi_channel_disable(NRF_PPI, ppi_ch);` here instead. This would be consistent with `nrf_ppi_channel_enable` used above and also a bit lighter.

Reply:
Good catch, done